packages/kit/src/runtime/client/utils.js

```diff
@@ -133,10 +133,11 @@ export function get_link_info(a, base) {
 		!url ||
 		!!target ||
 		is_external_url(url, base) ||
-		(a.getAttribute('rel') || '').split(/\s+/).includes('external') ||
-		a.hasAttribute('download');
+		(a.getAttribute('rel') || '').split(/\s+/).includes('external');
 
-	return { url, external, target };
+	const download = url?.origin === location.origin && a.hasAttribute('download');
```

Comment:
is the origin check necessary or could it just be this?

```suggestion
	const download = a.hasAttribute('download');
```

Reply:
The origin check ensures it only triggers on valid download links.
If a user (for whatever weird reason) has an invalid link such as:
```html
<a href="https://not-the-same-origin.com/file" download>...</a>
```
The browser navigates to the URL instead of downloading the file. In this case, it could be useful to have `beforeNavigate()` run.